modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/MediaSessionManager.kt

```diff
@@ -285,7 +285,16 @@ class MediaSessionManager(
             .observeOn(Schedulers.io())
             .switchMap { state ->
                 val episodeSource =
-                    if (state.isEmpty) Observable.just(Optional.empty()) else episodeManager.observeEpisodeByUuidRx(state.episodeUuid).distinctUntilChanged(BaseEpisode.isMediaSessionEqual).map { Optional.of(it) }.toObservable()
+                    if (state.isEmpty) {
+                        Observable.just(Optional.empty())
+                    } else {
+                        episodeManager.observeEpisodeByUuidRx(state.episodeUuid)
+                            .distinctUntilChanged(BaseEpisode.isMediaSessionEqual)
+                            .map { Optional.of(it) }
+                            // if the episode is deleted from the database while playing catch the error and just return an empty state
+                            .onErrorReturn { Optional.empty() }
+                            .toObservable()
+                    }
```

Comment:
As part of clearing the data when the episode was deleted, it caused the Up Next and playback event monitoring to crash. This meant that after clearing the data it wasn't possible to play another episode. This change fixes that.